### DISCUSSION TITLE
Edge 边上的 Label 标签或文字如何实现在边上的拖拽移动？

### DISCUSSION THREAD

**Discussion by esmadrider** [antvis/X6#3729] [Ideas]
## 背景：X6 已支持标签，但没找到标签拖拽移动

在 x6 的官方示例 https://x6.antv.vision/en/examples/edge/label#render-with-svg 中，描述了如何在边上创建一个固定位置的标签。

![CleanShot 2023-07-04 at 14 58 54](https://github.com/antvis/X6/assets/19285461/2d6dcdfe-e71c-4dfc-829b-7e8a82f13125)

如上图所示，无法左右拖拽移动，只能改一些 distance 值来静态修改。

## 想要的效果 1 - 标签支持拖拽移动

![CleanShot 2023-07-04 at 16 55 03](https://github.com/antvis/X6/assets/19285461/9702668c-1f9a-4117-9e9d-478c6b6a544a)

我在 Draw IO 上模拟了想要的这个效果（但其实这个例子中 draw io 移动的也是个元素），x6 的标签能否如动画这样支持？

## 如果效果 1 不行，降级方案 - 文字拖拽移动能否可以？

![CleanShot 2023-07-04 at 16 58 11](https://github.com/antvis/X6/assets/19285461/82c6c6fe-7cf0-45d2-86b6-42acdbaef841)

## 其它有趣的案例

https://projectstorm.cloud/react-diagrams/?path=/story/simple-usage--demo-simple

![CleanShot 2023-07-04 at 17 02 14](https://github.com/antvis/X6/assets/19285461/e1869e28-4b37-4568-9fa2-7f1510608b69)

## 我还看到了如下 x6 相关例子，但不知怎么应用到此 issue 上

* [examples/edge/edge#custom-click-event](https://x6.antv.vision/en/examples/edge/edge#custom-click-event): 这个例子能看到，一个边上有一个自定义元素，点击后它会在边上通过动画移动。所以移动效果是可以做的？
* [examples/node/tool#editable](https://x6.antv.vision/en/examples/node/tool#editable): 这个例子能看到，我们可以双击边上的任意位置，来创建一个 txt 文字到边上。但这个文字如何拖拽移动？

**Comment by NewByVector** [comment]
可以在初始化画布的时候配置：
```ts
new Graph({
  interacting: {
    edgeLabelMovable: true,
  },
})
```
不过现在边是可以自由移动的（没有限制在线上移动），这个已经加入到我们的工作计划中。

**Reply by esmadrider** [reply]
> 不过现在边是可以自由移动的（没有限制在线上移动），这个已经加入到我们的工作计划中。

是的，我打开 edgeLabelMovable 后，如下图，发现可以在边上自由移动，也就是 distant 值已经限制到了 0~1。（这应该是由于 edge 是一个矩形而被限制）

![CleanShot 2023-07-11 at 17 15 55](https://github.com/antvis/X6/assets/19285461/0c1a2dc1-021d-4231-9472-60289c1e8d72)

但是能看出来，offset 值没被限制，可以是负无穷到正无穷。也就是整个 Label 左右移动是受限制的，但上下移动会不受任何限制。

期待尽早能看到“限制到线上”移动功能。如果能给出一个 Roadmap 大致时间和大致版本就更好了。

既然这样，我在 issues 区提一个 feat issue 吧。同时把文字的拖拽移动提一下。期待解答，得尽快用了。

**Reply by liyulinfbfhcv** [reply]
希望可以尽快实现限制在线上移动
